test-case/check-userspace-paplay.sh

```diff
@@ -0,0 +1,113 @@
+OPT_PARM_lst['r']=1         OPT_VALUE_lst['r']=3
+
+OPT_OPT_lst['d']='duration' OPT_DESC_lst['d']='paplay duration in second'
+OPT_PARM_lst['d']=1         OPT_VALUE_lst['d']=8
+
+OPT_OPT_lst['f']='file'   OPT_DESC_lst['f']='source file path'
+OPT_PARM_lst['f']=1         OPT_VALUE_lst['f']='/dev/zero'
+
+OPT_OPT_lst['F']='format'   OPT_DESC_lst['F']='sample format'
+OPT_PARM_lst['F']=1         OPT_VALUE_lst['F']=s16le
+
+OPT_OPT_lst['R']='rate'   OPT_DESC_lst['R']='sample rate'
+OPT_PARM_lst['R']=1         OPT_VALUE_lst['R']=44100
+
+OPT_OPT_lst['C']='channels'   OPT_DESC_lst['C']='channels'
+OPT_PARM_lst['C']=1         OPT_VALUE_lst['C']=2
+
+func_opt_parse_option "$@"
+
+round_cnt=${OPT_VALUE_lst['r']}
+duration=${OPT_VALUE_lst['d']}
+file=${OPT_VALUE_lst['f']}
+format=${OPT_VALUE_lst['F']}
+rate=${OPT_VALUE_lst['R']}
+channel=${OPT_VALUE_lst['C']}
+
+[[ -e $file ]] || { dlogw "$file does not exist, use /dev/zero as dummy playback source" && file=/dev/zero; }
+
+# TODO: check the parameter is valid or not
+
+# go through all the sinks
+# get all the sinks name
+sinkkeys=$(pactlinfo.py --showsinks)
+for round in $(seq 1 $round_cnt); do
+    for i in $sinkkeys; do
+        sinkcard=$(pactlinfo.py --getsinkcardname "$i") || {
+            dlogw "failed to get sink $i card_name"
+            continue
+        }
+
+        # Let's skip testing on USB card
+        # TODO: add a list for other skipped cards such as HDA
+        if echo "$sinkcard" |grep -i "usb" &>/dev/null; then
+            continue
+        fi
+
+        # get the sink's active port
+        actport=$(pactlinfo.py --getsinkactport "$i") || {
+            dlogw "failed to get sink $i active port"
+            continue
+        }
+
+        # get the active port's information
+        actportvalue=$(pactlinfo.py --getsinkportinfo "$actport") || {
+            dlogw "failed to get sink $i active port $actport info"
+            continue
+        }
+
+        # check the active port is available or not from the port's information
+        portavailable=$(echo "$actportvalue" |grep "not available") || true
+        if [ -z "$portavailable" ]; then
+            # now prepare to paplay on this sink as the active port is not "not available"
+            # get the sink's name
+            sinkname=$(pactlinfo.py --getsinkname "$i")
+            sinkname=$(eval echo "$sinkname") #formalize the sink name
+            dlogi "===== Testing: (Round: $round/$round_cnt) (sink: $sinkname.$actport) ====="
+            dlogc "paplay -v --device=$sinkname --raw --rate=$rate --format=$format --channels=$channel $file"
+            paplay -v --device="$sinkname" --raw --rate=$rate --format=$format --channels=$channel $file &
+            pid=$!
+            sleep $duration
+            # check whether process is still running
+            count=$(ps -A| grep -c $pid) || true
+            if [[ $count -eq 0 ]]; then
+                if wait $pid; then #checks if process executed successfully or not
```

Comment:
The `ps | grep` is awkward and the sleep is probably inefficient. You want something much simpler like this:
```shell
timeout -k $((2*duration)) $duration paplay ...  &
paplayPID=$!
if wait $pid; then
```
This will get rid of the only shellcheck warning.

Reply:
In my original design, we have such scenarios based on the argument from user:
1. paplay stops successfully before timeout. This is good.
2. paplay try to stop but fails before timeout. This is wrong.
3.  paplay is timeouted and is killed by the command `kill`. This is still good.
If we use `timeout` command directly, I'm afraid we may miss the second scenario. 

I will submit the updated code firstly and let this discussion open. We can continue talking about this open in the updated version. Thanks for all the reviews @marc-hb